crates/task-impls/src/network.rs

```diff
@@ -102,109 +102,106 @@ pub struct NetworkMessageTaskState<TYPES: NodeType> {
+                        }
+
+                        GeneralConsensusMessage::ViewSyncFinalizeVote(view_sync_message) => {
+                            HotShotEvent::ViewSyncFinalizeVoteRecv(view_sync_message)
+                        }
+                        GeneralConsensusMessage::ViewSyncFinalizeCertificate(view_sync_message) => {
+                            HotShotEvent::ViewSyncFinalizeCertificate2Recv(view_sync_message)
+                        }
+
+                        GeneralConsensusMessage::TimeoutVote(message) => {
+                            HotShotEvent::TimeoutVoteRecv(message)
+                        }
+                        GeneralConsensusMessage::UpgradeProposal(message) => {
+                            HotShotEvent::UpgradeProposalRecv(message, sender)
+                        }
+                        GeneralConsensusMessage::UpgradeVote(message) => {
+                            error!("Received upgrade vote!");
+                            HotShotEvent::UpgradeVoteRecv(message)
+                        }
+                    },
+                    SequencingMessage::Da(da_message) => match da_message {
+                        DaConsensusMessage::DaProposal(proposal) => {
+                            HotShotEvent::DaProposalRecv(proposal, sender)
+                        }
+                        DaConsensusMessage::DaVote(vote) => HotShotEvent::DaVoteRecv(vote.clone()),
+                        DaConsensusMessage::DaCertificate(cert) => {
+                            HotShotEvent::DaCertificateRecv(cert)
+                        }
+                        DaConsensusMessage::VidDisperseMsg(proposal) => {
+                            HotShotEvent::VidShareRecv(proposal)
+                        }
+                    },
+                };
+                // TODO (Keyao benchmarking) Update these event variants (similar to the
+                // `TransactionsRecv` event) so we can send one event for a vector of messages.
+                // <https://github.com/EspressoSystems/HotShot/issues/1428>
+                broadcast_event(Arc::new(event), &self.internal_event_stream).await;
+            }
 
-                MessageKind::External(data) => {
-                    // Send the external message to the external event stream so it can be processed
+            // Handle data messages
+            MessageKind::Data(message) => match message {
+                DataMessage::SubmitTransaction(transaction, _) => {
                     broadcast_event(
-                        Event {
-                            view_number: TYPES::Time::new(1),
-                            event: EventType::ExternalMessageReceived(data),
-                        },
-                        &self.external_event_stream,
+                        Arc::new(HotShotEvent::TransactionsRecv(vec![transaction])),
+                        &self.internal_event_stream,
```

Comment:
I guess this was the whole reason for the batching of messages, to collect transactions.  Doesn't seem worth keeping to me

Reply:
I think it was also to potentially increase throughput by decreasing context switching, but we were not really making use of it in the way we could have. If we end up having better facilities in later for batching (e.g. a message type that is a vec of messages), I might consider reintroducing some form of batching